qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -45,6 +50,90 @@ class UndoEntry:
     pinned = attr.ib()
 
 
+class TabDeque:
+    """Class which manages the 'last visited' tab stack.
+
+    Instead of handling deletions by clearing old entries, they are handled by
+    checking if they exist on access. This allows us to save an iteration on
+    every tab delete.
+
+    Currently, we assume we will switch to the tab returned by any of the
+    getter functions. This is done because the on_switch functions will be
+    called upon switch, and we don't want to duplicate entries in the stack.
+    """
+
+    def __init__(self) -> None:
+        self._stack = collections.deque(
+            maxlen=config.val.tabs.focus_stack_size
+        )  # type:typing.Deque[weakref.ReferenceType[QWidget]]
+        self._stack_overflow = [
+        ]  # type: typing.List[weakref.ReferenceType[QWidget]]
+        self._overflow_special = ""
+
+    def on_switch(self, raw_tab: QWidget) -> None:
```

Comment:
Perhaps it would be better to note that the `raw_tab` parameter is the old tab (originally I thought it's the new one)